### PR TITLE
PE-4486: fallback to HEAD request for missing Content-Type

### PR DESCRIPTION
**Pull request by matibat**
Related PR: #1271

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/2tn2g35cruh98

### CODE REVIEW

**Review by thiagocarvalhodev — COMMENTED** [review]
I have the same question as @arielmelendez. Besides that, it looks good

File: lib/blocs/pin_file/file_id_resolver.dart [hunk 1]
```diff
@@ -67,10 +67,38 @@ class NetworkFileIdResolver implements FileIdResolver {
...
+      );
+      final response = await httpClient.head(uri);
+
+      final Map headers = response.headers;
+      final String? contentTypeHeader = headers['content-type'];
+
+      if (response.statusCode != 200 || contentTypeHeader == null) {
+        throw FileIdResolverException(
+          id: dataTxId,
+          cancelled: false,
```
[the remark this answers is not in the document]
Comment (matibat) [reply lib/blocs/pin_file/file_id_resolver.dart:84]: Indeed I'd make these to be different errors.

File: lib/blocs/pin_file/file_id_resolver.dart [hunk 2]
```diff
@@ -67,10 +67,38 @@ class NetworkFileIdResolver implements FileIdResolver {
     final _OwnerPrivacySizeAndType metadataTxInfo =
         await _getOwnerAndPrivacyOfDataTransaction(dataTxId);
 
+    String type;
+
+    if (metadataTxInfo.type == null) {
+      final uri = Uri.parse(
+        '${configService.config.defaultArweaveGatewayUrl}/$dataTxId',
+      );
+      final response = await httpClient.head(uri);
```
Comment (matibat) [inline lib/blocs/pin_file/file_id_resolver.dart:76]: Shouldn't retry?

File: lib/blocs/pin_file/file_id_resolver.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (matibat) [reply lib/blocs/pin_file/file_id_resolver.dart:84]: These booleans are confusing. A nice approach may be to add factories for it, e.g.: `FileIdResolverException.entityNotFound` and `FileIdResolverException.networkError`

**Review by matibat — COMMENTED** [review]
Self-review checkpoint

File: lib/blocs/pin_file/file_id_resolver.dart (hunk 2 above)
Comment (matibat) [reply lib/blocs/pin_file/file_id_resolver.dart]: Yes. Applied

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM

**Pull request merged**